Libraries/src/Amazon.Lambda.Annotations.SourceGenerator/CloudFormationTemplateHandler.cs

```diff
@@ -33,23 +42,63 @@ public string DetermineProjectRootDirectory(string sourceFilePath)
             return string.Empty;
         }
 
-        public string FindCloudFormationTemplate(string projectRootDirectory)
+        /// <summary>
+        /// Determines the path to the AWS serverless template file.
+        /// If the file does not exist then an empty file is created before returning the path.
+        /// </summary>
+        public string FindTemplate(string projectRootDirectory)
         {
-            var templateAbsolutePath = DetermineCloudFormationTemplatePath(projectRootDirectory);
+            var templateAbsolutePath = DetermineTemplatePath(projectRootDirectory);
 
             if (!_fileManager.Exists(templateAbsolutePath))
                 _fileManager.Create(templateAbsolutePath).Close();
-            
+
             return templateAbsolutePath;
         }
 
-        public bool DoesCloudFormationTemplateExist(string projectRootDirectory)
+        /// <summary>
+        /// Checks if the AWS serverless template file exists.
+        /// </summary>
+        public bool DoesTemplateExist(string projectRootDirectory)
         {
-            var templateAbsolutePath = DetermineCloudFormationTemplatePath(projectRootDirectory);
+            var templateAbsolutePath = DetermineTemplatePath(projectRootDirectory);
             return _fileManager.Exists(templateAbsolutePath);
         }
 
-        private string DetermineCloudFormationTemplatePath(string projectRootDirectory)
+        /// <summary>
+        /// Determines the file format of the AWS serverless template.
+        /// If the template does not exist or if the template is empty, then by default <see cref="CloudFormationTemplateFormat.Json"/> is returned.
+        /// </summary>
+        public CloudFormationTemplateFormat DetermineTemplateFormat(string templatePath)
+        {
+            if (!_fileManager.Exists(templatePath))
+            {
+                return CloudFormationTemplateFormat.Json;
+            }
+
+            var content = _fileManager.ReadAllText(templatePath);
+            if (string.IsNullOrEmpty(content))
+            {
+                return CloudFormationTemplateFormat.Json;
+            }
+
+            try
+            {
+                JsonConvert.DeserializeObject(content);
```

Comment:
Since a JSON CloudFormation template has to start with `{` and a YAML template would not start with a `{` in the lambda tooling we just look to see if the template starts with `{` https://github.com/aws/aws-extensions-for-dotnet-cli/blob/0d788f1015acbe836c4a4d8961f50e07c7924076/src/Amazon.Lambda.Tools/LambdaUtilities.cs#L355

With this logic you would switch to YAML if they had a JSON formatting issue. Plus it will take longer to parse the whole template.